vm/src/obj/objfloat.rs

```diff
@@ -574,6 +540,59 @@ impl PyFloat {
     }
 }
 
+#[derive(FromArgs)]
+struct FloatOptions {
+    #[pyarg(positional_only, optional = true)]
+    val_options: OptionalArg<PyObjectRef>,
+}
```

Comment:
I don't think this separate options struct is necessary, it's positional by default when you just use `OptionalArg` as a function argument.

Reply:
I actually had no idea you could do that, neat!